test/functional/wallet_hd.py

```diff
@@ -56,6 +56,12 @@ def start_and_connect_node1(self):
         connect_nodes(self.nodes[1], 0)
         self.sync_all()
 
+    def check_addressbook(self, old_book, new_book):
+        assert_equal(len(old_book), len(new_book))
+        for add in new_book:
+            assert add in old_book
+            assert_equal(old_book[add], new_book[add])
```

Comment:
Isn't this going to fail when we deal with an address that previously had a label in the addressbook and after the restore it doesn't?

Reply:
Sure. But we don't set any label in this test.
The reason why I added this line is to check that the *purpose* is the same.